doc/plugins.dox

```diff
@@ -326,27 +426,33 @@ For the moment you don't need to worry about the parameters of that function.
 
 Now we need to clean up properly when the plugin is unloaded.
 
-To remove the menu item from the Tools menu, you can use gtk_widget_destroy().
-gtk_widget_destroy() expects a pointer to a GtkWidget object.
+To remove the menu item from the Tools menu you can use gtk_widget_destroy().
 
-First you should add gtk_widget_destroy() to your plugin_cleanup() function.
-The argument for gtk_widget_destroy() is the widget object you created earlier in
-plugin_init(). To be able to access this pointer in plugin_cleanup(), you need to move
-its definition from plugin_init() into the global context so its visibility will increase
-and it can be accessed in all functions.
-@code
-static GtkWidget *main_menu_item = NULL;
+First you should add gtk_widget_destroy() to your GeanyPluginFuncs::cleanup() function. The
+argument for gtk_widget_destroy() is the widget object you created earlier in
+GeanyPluginFuncs::init(). To be able to access this pointer in GeanyPluginFuncs::cleanup() you can
+use geany_plugin_set_data() to set plugin-defined data pointer to the widget. Alternatively, you
+can store the pointer in some global variable so its visibility will increase and it can be
+accessed in all functions.
 
+@code
+/* alternative: global variable:
+static GtkWidget *main_menu_item;
+*/
 // ...
-void plugin_init(GeanyData *data)
+static gboolean hello_init(GeanyPlugin *plugin, gpointer pdata)
 {
-	main_menu_item = gtk_menu_item_new_with_mnemonic("Hello World");
+	GtkWidget *main_menu_item = gtk_menu_item_new_with_mnemonic("Hello World");
 	gtk_widget_show(main_menu_item);
 // ...
+	geany_plugin_set_data(plugin, main_menu_item, NULL);
+	retrun TRUE;
```

Comment:
just a leftover because the diff I commented on didn't change when kugel fixed it